source/common/router/config_impl.cc

```diff
@@ -464,9 +464,11 @@ RouteMatcher::RouteMatcher(const Json::Object& config, Runtime::Loader& runtime,
 RouteConstSharedPtr VirtualHostImpl::getRouteFromEntries(const Http::HeaderMap& headers,
                                                          uint64_t random_value) const {
   // First check for ssl redirect.
-  if (ssl_requirements_ == SslRequirements::ALL && headers.ForwardedProto()->value() != "https") {
+  if (ssl_requirements_ == SslRequirements::ALL && headers.ForwardedProto() &&
```

Comment:
Prefer `headers.ForwardedProto() != nullptr` for better readability (it's clear we're dealing with a pointer).